examples/twittermap/web/public/javascripts/map/countmap.js

```diff
@@ -128,8 +334,15 @@ angular.module('cloudberry.map')
             // Since the middleware returns the query result in multiple steps,
             // cloudberryService.timeSeriesQueryResult stores the current intermediate result.
             cloudberry.timeSeriesQueryResult = resultSet[0];
+
             // Avoid memory leak.
             resultSet[0] = [];
+            $scope.ChartDataMap = concatHashmap(
+              TimeSeriesCache.arrayToStore(cloudberry.parameters.geoIds,cloudberry.timeSeriesQueryResult,cloudberry.parameters.geoLevel),
+              $scope.ChartDataMap
+            );
```

Comment:
As an affect of DRUM, large query results are returned in sections. Thus before the entire timeseries data is obtained, we have: `$scope.ChartDataMap = newest/current subset of query result + cached timeseries`
When query results are return in more than two sections, updating $scope.ChartDataMap with an elder version of itself will create duplicate copies of the cached timeseries portion. Change line342 and line403 as follow(in bold) will solve this problem:

> $scope.ChartDataMap = concatHashmap(
> TimeSeriesCache.arrayToStore(cloudberry.parameters.geoIds,cloudberry.timeSeriesQueryResult,cloudberry.parameters.geoLevel),
> **TimeSeriesCache.getInViewTimeSeriesStore(cloudberry.parameters.geoIds,cloudberry.parameters.timeInterval)**
> );

Reply:
The reason for line 342 `$scope.ChartDataMap = concatHashmap(
              TimeSeriesCache.arrayToStore(cloudberry.parameters.geoIds,cloudberry.timeSeriesQueryResult,cloudberry.parameters.geoLevel),
              $scope.ChartDataMap
            );`
is because in line 330 already call `$scope.chartDataMap = TimeSeriesCache.getInViewTimeSeriesStore(cloudberry.parameters.geoIds,cloudberry.parameters.timeInterval);` 
But it may cause duplicate copies. I will change the call and test it. Thank you!